File: hash/lv2_전화번호목록.java
// https://programmers.co.kr/learn/courses/30/lessons/42577?language=java
import java.util.Arrays;


class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        for (int i = 0; i < phone_book.length; i++) {
            for (int j = 0; j < phone_book.length; j++) {
                if (i != j && phone_book[j].startsWith(phone_book[i])) {
                    return false;
                }
            }
        }
        return answer;
    }
}

File: hash/lv2_전화번호목록.py
# https://programmers.co.kr/learn/courses/30/lessons/42577?language=python3

def solution(phone_book):
    phone_book.sort()
    for p_ in phone_book:
        for p__ in phone_book:
            if p_ != p__ and p__.startswith(p_):
                return False
    return True

